packages/zoe/src/contractFacet/offerHandlerStorage.js

```diff
@@ -0,0 +1,27 @@
+import { makeWeakStore as makeNonVOWeakStore } from '@agoric/store';
+
+import { makeHandle } from '../makeHandle';
+
+export const makeOfferHandlerStorage = () => {
+  /** @type {WeakStore<InvitationHandle, (seat: ZCFSeat) => unknown>} */
+  const invitationHandleToHandler = makeNonVOWeakStore('invitationHandle');
+
+  /** @type {(offerHandler: OfferHandler) => InvitationHandle} */
+  const storeOfferHandler = offerHandler => {
+    const invitationHandle = makeHandle('Invitation');
+    invitationHandleToHandler.init(invitationHandle, offerHandler);
+    return invitationHandle;
+  };
+
+  /** @type {(invitationHandle: InvitationHandle) => OfferHandler} */
+  const getOfferHandler = invitationHandle => {
```

Comment:
```suggestion
  const takeOfferHandler = invitationHandle => {
```

This `getFoo` operation is destructive. The Foo can only be gotten once. For this, I use the convention `takeFoo`. Goes back to the old Agorics days, actually.

Reply:
Sounds good, I like that.

Reply:
Done